src/components/views/messages/MPollBody.tsx

```diff
@@ -43,74 +44,93 @@ import PollCreateDialog from "../elements/PollCreateDialog";
 import { MatrixClientPeg } from "../../../MatrixClientPeg";
 
 interface IState {
-    selected?: string; // Which option was clicked by the local user
+    selected?: string | null | undefined; // Which option was clicked by the local user
     voteRelations: RelatedRelations; // Voting (response) events
     endRelations: RelatedRelations; // Poll end events
 }
 
 export function createVoteRelations(
-    getRelationsForEvent: (
-        eventId: string,
-        relationType: string,
-        eventType: string
-    ) => Relations,
+    getRelationsForEvent: GetRelationsForEvent,
     eventId: string,
 ) {
-    return new RelatedRelations([
-        getRelationsForEvent(
-            eventId,
-            "m.reference",
-            M_POLL_RESPONSE.name,
-        ),
-        getRelationsForEvent(
-            eventId,
-            "m.reference",
-            M_POLL_RESPONSE.altName,
-        ),
-    ]);
+    const relationsList: Relations[] = [];
+
+    const pollResponseRelations = getRelationsForEvent(
+        eventId,
+        "m.reference",
+        M_POLL_RESPONSE.name,
+    );
+    if (pollResponseRelations) {
+        relationsList.push(pollResponseRelations);
+    }
+
+    const pollResposnseAltRelations = getRelationsForEvent(
+        eventId,
+        "m.reference",
+        M_POLL_RESPONSE.altName,
+    );
+    if (pollResposnseAltRelations) {
+        relationsList.push(pollResposnseAltRelations);
+    }
+
+    return new RelatedRelations(relationsList);
 }
 
 export function findTopAnswer(
     pollEvent: MatrixEvent,
     matrixClient: MatrixClient,
-    getRelationsForEvent?: (
-        eventId: string,
-        relationType: string,
-        eventType: string
-    ) => Relations,
+    getRelationsForEvent?: GetRelationsForEvent,
 ): string {
     if (!getRelationsForEvent) {
         return "";
     }
 
+    const pollEventId = pollEvent.getId();
+    if (!pollEventId) {
+        logger.warn(
+            "findTopAnswer: Poll event needs an event ID to fetch relations in order to determine " +
+            "the top answer - assuming no best answer",
+        );
+        return "";
+    }
```

Comment:
I also don't have interest in adding tests for these safety checks to get SonarCloud happy with code coverage, https://sonarcloud.io/summary/new_code?id=matrix-react-sdk&pullRequest=9558

Reply:
On the other hand: Having tests before fixing strict type things would give us confidence that it is still working (as expected).

Reply:
Discussed out of band in the [Element web internal room](https://matrix.to/#/!fLeHeojWgBGJlLNdLC:matrix.org/$pu3_zDtI8oE_9gR6iavnBb-7Swj7ZVjkSkPi8cETDYw?via=matrix.org&via=element.io&via=vector.modular.im) with @t3chguy about getting this PR merged but it was said that 80% Sonar code coverage is required.

I'd rather focus on adding coverage with test cases that matter to the codebase. Not sanity type check drivel that Sonar is begging for here.

I could have more easily added `!` but that gives no value for properly handling things. This PR actually tries to handle most of the cases with application failure modes instead of undefined behavior.

Merging something under 80% according to Sonar lowers the `develop` average but our [overall coverage is 41.8%](https://sonarcloud.io/component_measures?metric=coverage&view=list&id=matrix-react-sdk)  so there isn't even a target threshold to worry about dipping. And even if we were at a goal threshold, we would probably have enough margin to let this PR through and still be at 80% overall.

Back to the main point, "I'd rather focus on adding coverage with test cases that matter to the codebase. Not sanity type check drivel that Sonar is begging for here." We can increase coverage with meaningful tests elsewhere.

The conclusion from the discussion there is basically that I should show up to the Web Weekly meeting to discuss how to get exempt from the coverage gate as there are no rules for it now.

---

Aside: It's totally unclear that tsc strict linter is allowed to fail (until next week) which spawned these changes. I suppose this is mostly because GitHub actions don't have an "allowed-to-fail" behavior where the icon changes to `!`. But I assume we could switch it to pass and still add the lints to the diff. And would be nice to echo out a message in that job that these are allowed to fail or "not absolutely necessary to make this happy" (add some ANSI color to make it stand out at the top and bottom). But this point doesn't even matter because the strict TypeScript checks will become required next week anyway so this change would happen if I just waited another week.

The `strict` errors are also extremely hard to diagnose locally since there it's not in any config or `package.json` `script` to run. The solution here is better tooling or solving all of the `strict` errors on `develop` so we don't run into unrelated problems to the PR at hand. @robintown mentioned installing `typescript-strict-plugin` but it doesn't have the diff comparison and unclear how easy to turn it on/off.

Reply:
> But I assume we could switch it to pass and still add the lints to the diff. And would be nice to echo out a message in that job that these are allowed to fail or "not absolutely necessary to make this happy" (add some ANSI color to make it stand out at the top and bottom).

If it is green (passing) then no one would have any reason to open the CI log to see the pretty ANSI colours though, no?

Reply:
> If it is green (passing) then no one would have any reason to open the CI log to see the pretty ANSI colours though, no?

At the very minimum in any case, if the tsc strict job is optional, then it should have the ANSI colored notes in the job about it being optional.

In terms of whether we should make the tsc strict job green (passing) since it's optional; The comments can still appear in the diff and people can check that job out to get a full list if they see one in the diff.

Ex.
<img src="https://user-images.githubusercontent.com/558581/200919196-96fc9639-337e-402e-beba-571d826e1e93.png" width="396"> 

But all of this doesn't matter since it will be required next week and none of this matters to change in that case. I suggest we just hide these two comments because of this.

Reply:
Discussed in the [Web Weekly meeting today](https://docs.google.com/document/d/1wa1jGYsN-rOzvQB4tky5e3lwy3jl5PNEVneALwwg8XM/edit#bookmark=id.eqcq6590kpfi) and team is okay with letting this code coverage exception through ✅

Especially since it's at 76% coverage and sentiment agreement about there being better uses of our time to add more valuable tests elsewhere.